prime-dotnet-webapi/Controllers/OrganizationsController.cs

```diff
@@ -254,5 +259,96 @@ public async Task<ActionResult<Organization>> SubmitOrganizationRegistration(int
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiResultResponse<Organization>), StatusCodes.Status201Created)]
+        public async Task<ActionResult<SignedAgreement>> CreateSignedAgreement(int organizationId, [FromQuery] Guid documentGuid, [FromQuery] string filename)
+        {
+            var organization = await _organizationService.GetOrganizationAsync(organizationId);
+
+            if (organization == null)
+            {
+                return NotFound(ApiResponse.Message($"Organization not found with id {organizationId}"));
+            }
+
+            if (!User.CanEdit(organization.SigningAuthority))
+            {
+                return Forbid();
+            }
+
+            var licence = await _organizationService.AddSignedAgreementAsync(organization.Id, documentGuid, filename);
+
+            // TODO updated to be licence instead of organization, and should have GET and CreatedAtAction
+            return Ok(ApiResponse.Result(licence));
+            // return CreatedAtAction(
+            //     nameof(GetOrganizationById),
+            //     new { organizationId = createdOrganizationId },
+            //     ApiResponse.Result(createdOrganization)
+            // );
+        }
+
+        // Get: api/organizations/5/signed-agreement
+        /// <summary>
+        /// Gets the signed agreement for a organization.
+        /// </summary>
+        /// <param name="organizationId"></param>
+        [HttpGet("{organizationId}/signed-agreement", Name = nameof(GetSignedAgreement))]
+        [ProducesResponseType(typeof(ApiBadRequestResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiResultResponse<Organization>), StatusCodes.Status201Created)]
+        public async Task<ActionResult<IEnumerable<SignedAgreement>>> GetSignedAgreement(int organizationId)
+        {
+            var organization = await _organizationService.GetOrganizationAsync(organizationId);
+
+            if (organization == null)
+            {
+                return NotFound(ApiResponse.Message($"Organization not found with id {organizationId}"));
+            }
+
+            if (!User.CanEdit(organization.SigningAuthority))
+            {
+                return Forbid();
+            }
+
+            var licences = await _organizationService.GetSignedAgreementsAsync(organization.Id);
+
+            return Ok(ApiResponse.Result(licences));
+        }
+
+        // GET: api/Organizations/download-organization-agreement
+        /// <summary>
+        /// Download the organization agreement document.
+        /// </summary>
+        [HttpGet("download-organization-agreement", Name = nameof(DownloadOrganizationAgreement))]
```

Comment:
REST endpoints don't generally have verbs in them. How about GET Organizations/organization-agreement-document ?

Reply:
I could also see this being on the document controller, so it would be GET document/organization-agreement

Reply:
That's a great idea James, and I think in the next sprint we should definitely create a document controller.